### PR TITLE
add an upload-assets-to-s3 helper 🐿 v2.12.5

### PR DESCRIPTION
**Pull request by pixelandpage**
Adds a helper to upload the generated assets for an app to S3 using [the awscli helper](https://github.com/Financial-Times/next-ci-shared-helpers/pull/22)

This is required along with https://github.com/Financial-Times/next-ci-shared-helpers/pull/22 to remove the dependency on n-heroku-tools when deploying assets for Page Kit apps. 

Related n-gage changes: https://github.com/Financial-Times/n-gage/pull/213

Related issue on Page Kit: https://github.com/Financial-Times/next-ci-shared-helpers/issues/21

### CODE REVIEW

File: helper-upload-assets-to-s3 [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+		--acl="public-read"
+}
+
+upload_files() {
+	local SOURCE_FILES=($(find public/*.{js,css,gz,br,map} 2>/dev/null))
+
+	for FILE in ${SOURCE_FILES[@]}; do
+		upload_file "$FILE" $DESTINATION_BUCKET_EU
+		upload_file "$FILE" $DESTINATION_BUCKET_US
+	done
```
[the remark this answers is not in the document]
Comment (sjparkinson) [reply helper-upload-assets-to-s3]: Correct, but I suggest we move away from the dependency on that.

If we upload to both as part of the pipeline, and they the two uploads are both a success, there is no need for the poller that waits for AWS to replicate the objects for us.

[the remark this answers is not in the document]
Comment (i-like-robots) [reply helper-upload-assets-to-s3]: Can we add a comment to this code which states this?

[the remark this answers is not in the document]
Comment (pixelandpage) [reply helper-upload-assets-to-s3]: yes 👍

File: helper-upload-assets-to-s3 [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+	aws s3 cp $FILE "s3://$DESTINATION_BUCKET/$DESTINATION_FOLDER/$FILE" \
+		--cache-control=31536000 \
+		--content-type="$TYPE; charset=utf-8" \
+		--content-encoding="$ENCODING" \
+		--acl="public-read"
+}
+
+upload_files() {
+	local SOURCE_FILES=($(find public/*.{js,css,gz,br,map} 2>/dev/null))
```
[the remark this answers is not in the document]
Comment (pixelandpage) [reply helper-upload-assets-to-s3]: Cool, I think I've resolved this now

File: helper-upload-assets-to-s3 [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+# Set error handling
+set -x -eu -o pipefail
+
+# HELPER COMMANDS
+
+
+DESTINATION_BUCKET_EU="ft-next-hashed-assets-prod"
+DESTINATION_BUCKET_US="ft-next-hashed-assets-prod-us"
+DESTINATION_FOLDER="hashed-assets/page-kit"
+SOURCE_FOLDER="public"
```
Comment (i-like-robots) [inline helper-upload-assets-to-s3]: I think `DESTINATION_FOLDER` and `SOURCE_FOLDER` should be args but these could be the default values, e.g.:

```bash
DESTINATION_FOLDER=${2:-"hashed-assets/page-kit"}
SOURCE_FOLDER=${1:-"public"}
```

**Pull request merged**